spec/controllers/books_controller_spec.rb

```diff
@@ -2,4 +2,118 @@
 
 RSpec.describe BooksController, type: :controller do
 
+  let(:user) {
+    create :user
+  }
+
+  let(:book) {
+    create :book, title: 'first', user: user
+  }
+
+  let(:valid_attributes) {
+    {
+      :title => "ValidTitle",
+      :user => user
+    }
+  }
+
+  let(:valid_session) { {} }
+
+  describe "GET #index" do
+    it "assigns all books as @books" do
+      expect(book.save).to be(true)
+
+      @token = AuthenticateUser.call(user.email, user.password)
+      request.headers["Authorization"] = @token.result
+
+      get :index
+      expect(assigns(:books)).to eq([book])
+    end
+  end
+
+  describe "GET #show" do
+    it "assigns book as @book" do
+      expect(book.save).to be(true)
+
+      @token = AuthenticateUser.call(user.email, user.password)
+      request.headers["Authorization"] = @token.result
+
+      get :show, :params => {id_book: book.id, id: user.id}
+      expect(assigns(:book)).to eq(book)
+    end
+  end
+
+  describe "PUT #update" do
+
+
+
+    context "with valid params" do
+
+      let(:new_attributes) {
+        {:title => "newValid Title"}
+      }
+
+      let(:invalid_attributes) {
+        {:title => "ivT"}
+      }
+
+      it "updates the requested @book" do
+        book = FactoryGirl.create :book
+        @token = AuthenticateUser.call(book.user.email, book.user.password)
+        request.headers["Authorization"] = @token.result
+        put :update, params: {id: book.id, book: new_attributes}, session: valid_session
+        book.reload
+        expect(book.title).to eq(new_attributes[:title])
+      end
+
+      context "with invalid params" do
+        it "assigns the book as @book" do
+          book = FactoryGirl.create :book
+          @token = AuthenticateUser.call(book.user.email, book.user.password)
+          request.headers["Authorization"] = @token.result
+          put :update, params: {id: book.id, book: invalid_attributes}, session: valid_session
+          expect(assigns(:book)).to eq(book)
+        end
```

Comment:
Esse teste passa mas está errado. Ele deveria testar se usuário não é atualizado com parâmetros inválidos.

1º O nome do teste está errado. Setar book é coisa de "#show", no update com parâmetros inválidos esse texto deveria ser algo como "Não atualiza um book com parâmetros inválidos".
2º Envia parâmetros inválidos no "put" mas faz "expect(assigns(:book)).to eq(book)" é como fazer "1 == 1". Deveria verificar os atributos de book não foram atualizados.

Veja como o teste de update do user faz